app/src/androidTest/java/com/example/rwothoromo/developers/MainActivityTest.java

```diff
@@ -0,0 +1,173 @@
+            Intent intent = new Intent(targetContext, MainActivity.class);
+            return intent;
+        }
+    };
+
+    private CountingIdlingResource countingResource;
+
+    /**
+     * Register any resource that needs to be synchronized with Espresso before the test is run.
+     *
+     * @throws Exception when method fails
+     */
+    @Before
+    public void setUp() throws Exception {
+        countingResource = new CountingIdlingResource("IDLING_RESOURCE");
+        Espresso.registerIdlingResources(countingResource);
+    }
+
+    /**
+     * Check that RecyclerView is displayed.
+     */
+    @Test
+    public void displaysRecyclerView() {
+        onView(withId(R.id.recyclerView)).check(matches(withEffectiveVisibility(
+                ViewMatchers.Visibility.VISIBLE)));
+    }
+
+    /**
+     * Displays the Toolbar.
+     */
+    @Test
+    public void displaysToolbar() {
+        onView(withId(R.id.toolbar)).check(matches(isDisplayed()));
+    }
+
+    /**
+     * View the MainActivity title.
+     */
+    @Test
+    public void displaysToolbarTitle() {
+        onView(withText(R.string.app_name)).check(matches(withParent(withId(R.id.toolbar))));
+    }
+
+    /**
+     * View the MainActivity subtitle.
+     */
+    @Test
+    public void displaysToolbarSubtitle() {
+        onView(withText(R.string.app_subtitle)).check(matches(withParent(withId(R.id.toolbar))));
+    }
+
+    /**
+     * Click the Search icon on the Action bar.
+     */
+    @Test
+    public void clickActionBarSearchItem() {
+        // Click on the Search icon.
+        onView(withId(R.id.action_search)).perform(click());
+    }
+
+    /**
+     * Click Settings in the menu items.
+     */
+    @Test
+    public void clickActionBarOverflowSettings() {
+        // Open the options menu OR open the overflow menu, depending on whether
+        // the device has a hardware or software overflow menu button.
+        openActionBarOverflowOrOptionsMenu(getInstrumentation().getTargetContext());
+
+        // Click the item.
+        onView(withText(R.string.action_settings)).perform(click());
+    }
+
+    /**
+     * Click Refresh in the menu items.
+     */
+    @Test
+    public void clickActionBarOverflowRefresh() {
+        // Open the options menu OR open the overflow menu, depending on whether
+        // the device has a hardware or software overflow menu button.
+        openActionBarOverflowOrOptionsMenu(getInstrumentation().getTargetContext());
+
+        // Click the item.
+        onView(withText(R.string.action_refresh)).perform(click());
+    }
+
+    /**
+     * Scroll to a GitHub user on the RecyclerView and click one.
+     */
+    @Test
+    public void clickableRecyclerViewItems() throws InterruptedException {
+
+        onView(withId(R.id.recyclerView)).check(matches(isDisplayed()));
+        Thread.sleep(5000);
+
+        Intents.init();
+
+        // Scroll to an item at a position and click on it.
+        final int mockPosition = 0;
```

Comment:
Remember our discussion about this, you cannot be so sure there will always be an item at this position (or any position at all) if your tests are dependent on network calls (or more precisely, unpredictable data).

Reply:
Oh yes, I'll try mocking this instead.

Reply:
I have updated the `IdlingResource` correctly, incremented it before the API and decremented it `onResponse` and `onFailure`.

All `UI tests` are now working.